src/tools/info/tl_info.c

```diff
@@ -334,7 +334,9 @@ static ucs_status_t print_tl_info(uct_md_h md, const char *tl_name,
     ucs_status_t status;
     unsigned i;
 
-    status = ucs_async_context_init(&async, UCS_ASYNC_MODE_THREAD);
+    status = ucs_async_context_init(&async, RUNNING_ON_VALGRIND ?
+                                    UCS_ASYNC_MODE_THREAD_MUTEX :
```

Comment:
I'd hide this `if` under some macro, because it is used several times

Reply:
done